client/js/lounge.js

```diff
@@ -361,14 +361,13 @@ $(function() {
 					let value = params[key];
 
 					if (key === "join") {
-						const channels = value.split(",");
-						value = channels.map((c) => {
-							if (c.match(/^\w/)) {
-								return "#" + c;
+						value = value.split(",").map((chan) => {
+							if (!chan.match(/^[#&!+]/)) {
+								return `#${chan}`;
```

Comment:
I actually like the idea that we could open a given query window from the `join` URL param.

I have seen a few personal websites where people set up a web chat that opens a query window with them (the website author) but that's just one use case. I can see how it could be useful to have default channels _and_ queries open by default.

Any chance we could keep that behavior? That was actually what #2327 was about rather than convert everything to a channel.

Reply:
That would break #2289.

Reply:
I guess it does, I didn't think of that at the time of that PR, which I now think we shouldn't have done.

I really don't want to get rid of the capability to auto-open queries though. I can understand how it's a pain to add # in the URL (still would prefer to open a query though), but in the join field of the connect window, I really think we shouldn't assume that `foo` actually means `#foo`. If no `#` is specified, it should open a query IMO.

Reply:
I think better over all to have a consistent behaviour where client and server agree, in this case to prefix the channel.

Honestly, I don't see much of a use to support opening queries there, the field is called `Channels`.

Reply:
Alright then. I do see a use for it (I even gave an actual use case for it, that I have actually seen out there), but I'm not going to push for it. We'll see if people create tickets for it :D